core/deployment/src/main/java/io/quarkiverse/operatorsdk/deployment/OperatorSDKProcessor.java

```diff
@@ -398,7 +398,7 @@ private QuarkusControllerConfiguration createControllerConfiguration(
                     getLabelSelector(controllerAnnotation),
                     Optional.ofNullable(specClassName),
                     Optional.ofNullable(statusClassName),
-                    dependentResources);
+                    dependentResources.values().stream().collect(Collectors.toUnmodifiableList()));
```

Comment:
Why not use directly a list instead of creating a map in the `createDependentResources` method?

Reply:
So that it's easier to check if a dependent resource with the same name is already configured.